sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GatewayAddressCache.java

```diff
@@ -411,7 +411,17 @@ public void updateAddresses(final URI serverKey) {
             if (!(exception instanceof CosmosException)) {
                 // wrap in CosmosException
                 logger.error("Network failure", exception);
-                dce = BridgeInternal.createCosmosException(request.requestContext.resourcePhysicalAddress, 0, exception);
+                int statusCode = 0;
+                if (WebExceptionUtility.isNetworkFailure(exception)) {
```

Comment:
any possibility the exception is not networkFailure? then we might still return statusCode 0?

Reply:
Yes - intentionally. Beczause at that point I can't come-up with a better status code - so don't want to change the behavior